app/src/main/java/com/quran/labs/androidquran/ui/util/ToastCompat.java

```diff
@@ -0,0 +1,83 @@
+package com.quran.labs.androidquran.ui.util;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.StringRes;
+import androidx.appcompat.content.res.AppCompatResources;
+import androidx.core.content.ContextCompat;
+import androidx.core.graphics.drawable.DrawableCompat;
+
+import com.quran.labs.androidquran.R;
+import com.quran.labs.androidquran.util.QuranSettings;
+
+public class ToastCompat {
+
+  private static final int ANDROID_R = 30;
+
+  public static Toast makeText(@NonNull Context context, @StringRes int messageRes, int duration) {
+    return makeText(context, context.getString(messageRes), duration);
+  }
+
+  public static Toast makeText(@NonNull Context context, CharSequence message, int duration) {
+    @SuppressLint("ShowToast") Toast toast = Toast.makeText(context, message, duration);
+    if (Build.VERSION.SDK_INT < ANDROID_R) {
+      // It's unclear what behavior `Toast.setView()` has on R and above, hence the guard.
+      toast.setView(createToastView(context, message));
```

Comment:
`Toast.getView()` and `Toast.setView()` are deprecated in Android R. I read (somewhere) that `getView()` will return `null`, and I don't know what the behaviour of `setView()` will be, though I wouldn't be surprised if it's a no-op on Android R (if `targetSdkVersion >= 30`).

I wasn't able to test on R, so I left this with a guard. This means that on R and above, unless this logic is amended, toasts will do whatever they do by default, which means that the toast will be light colored.

Reply:
if `Toast.getView()` is not deprecated until R and we're only supporting this R and below, could we not just let the normal Toast ui inflate itself, get its view, and if it's not null, apply a background instead?

Reply:
Yep we can, but as you mentioned, different device manufacturers could do odd things so we'd have to set the background and text color in any case.

Reply:
*we can but I don't recommend. If the device manufacturer inflates its own layout for example, we wouldn't know the ID for the textview.